pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CollectorGitHubDocument = "GitHubCollector"
+)
+
+type githubDocumentCollector struct {
+	poll     bool
+	interval time.Duration
+	token    string
+	owner    string
+	repo     string
+	tag      string
+	tagList  []string
+}
+
+func NewGitHubDocumentCollector(ctx context.Context, poll bool, interval time.Duration, logger *zap.SugaredLogger, token string, owner string, repo string, tag string, tagList []string) *githubDocumentCollector {
+	return &githubDocumentCollector{
+		poll:     poll,
+		interval: interval,
+		token:    os.Getenv("API_KEY"),
```

Comment:
i think this should not be using env to get this and should be an input to the func